### ISSUE TITLE
[HLSL] Enable HLSL availability diagnostic on exported library functions and add tests

### ISSUE THREAD

**Issue by hekota** [llvm/llvm-project#92073]
This is a tracking work item to enable HLSL availability diagnostic on exported library functions and to add tests. This can happen as soon as #96823 is resolved.

The diagnostic of exported function can be enabled in `DiagnoseHLSLAvailability::RunOnTranslationUnit`.

The tests should be added to the files in `clang\test\SemaHLSL\Availability` for all diagnostic modes. For example this is the minimum that should be added to default availability test for libraries:

```
// Exported function without body
export void exportedFunction(float f);

// Exported function with body
export void exportedFunction(float f) {
  // expected-error@#exportedFunction_fx_call {{'fx' is only available on Shader Model 6.5 or newer}}
  // expected-note@#fx {{'fx' has been marked as being introduced in Shader Model 6.5 here, but the deployment target is Shader Model 6.0}}
  float A = fx(f); // #exportedFunction_fx_call

  // API with shader-stage-specific availability in exported library function
  // - no errors expected because the actual shader stage this function
  // will be used in not known at this time
  float B = fy(f);
  float C = fz(f);
}
```
